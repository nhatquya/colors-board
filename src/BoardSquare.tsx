import React, { useRef } from "react";
import { useDrag, useDrop } from "react-dnd";
import { ITEM_TYPE } from "./constants";

export interface Position {
  x: number;
  y: number;
}

interface BoardSquareProps {
  position: Position;
  color: string;
  swapSquare: (source: Position, target: Position) => void;
}

const boardSquareStyle = (color: string, opacity: number) =>
  ({
    position: "relative",
    width: "12.5%",
    height: "12.5%",
    boxShadow: "inset 0px 0px 0px 0.3px #000",
    backgroundColor: color,
    opacity,
  } as React.CSSProperties);

const BoardSquare: React.FC<BoardSquareProps> = (props) => {
  const { color, position, swapSquare } = props;
  const ref = useRef(null);

  const [{ isDragging }, drag] = useDrag({
    item: { type: ITEM_TYPE.BOARD_SQUARE, data: position },
    collect: (monitor) => ({
      isDragging: !!monitor.isDragging(),
    }),
  });

  const [{ isOver }, drop] = useDrop({
    accept: ITEM_TYPE.BOARD_SQUARE,
    canDrop: (item: any) => item !== position,
    drop: (item: any) => {
      swapSquare(item.data, position);
    },
    collect: (monitor) => ({
      isOver: !!monitor.isOver(),
      canDrop: !!monitor.canDrop(),
    }),
  });

  const opacity = isDragging || isOver ? 0.5 : 1;
  drag(drop(ref));
  return (
    <div
      ref={ref}
      style={boardSquareStyle(color, opacity)}
      className="board-square"
    />
  );
};

export default BoardSquare;
