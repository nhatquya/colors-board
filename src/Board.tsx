import React, { useState } from "react";
import BoardSquare, { Position } from "./BoardSquare";
import { prepareBoardSquares } from "./utils";

const BOARD_SIZE = 8;

const boardStyle: React.CSSProperties = {
  width: "100%",
  height: "100%",
  display: "flex",
  flexWrap: "wrap",
};

export interface BoardSquare {
  x: number;
  y: number;
  color: string;
}

function renderSquare(
  square: BoardSquare,
  swapSquare: (source: Position, target: Position) => void
) {
  const { x, y, color } = square;
  return (
    <BoardSquare
      key={`x-${x}-y${y}`}
      position={{ x, y }}
      color={color}
      swapSquare={swapSquare}
    />
  );
}

const Board: React.FC = () => {
  const squares = prepareBoardSquares(BOARD_SIZE);
  const [boardSquares, setBoardSquares] = useState(squares);

  const swapSquare = (source: Position, target: Position) => {
    const sourceIndex = boardSquares.findIndex(
      (item) => item.x === source.x && item.y === source.y
    );
    const targetIndex = boardSquares.findIndex(
      (item) => item.x === target.x && item.y === target.y
    );
    const newBoardSquares = [...boardSquares];
    const tmpSquare = { ...newBoardSquares[sourceIndex] };
    newBoardSquares[sourceIndex].color = newBoardSquares[targetIndex].color;
    newBoardSquares[targetIndex].color = tmpSquare.color;
    setBoardSquares(newBoardSquares);
  };

  return (
    <div style={boardStyle}>
      {boardSquares.map((item) => renderSquare(item, swapSquare))}
    </div>
  );
};
export default Board;
