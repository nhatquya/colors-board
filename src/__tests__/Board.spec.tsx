import React from 'react'
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Board from '../Board';
import BoardSquare from '../BoardSquare';

jest.mock('../BoardSquare', () => (props: any) => <mock-board-square {...props} />);

describe('Board', () => {
    afterAll(() => {
        jest.resetAllMocks();
    })

    it('should render board squares correctly', () => {
        const root = mount(<Board />);
        expect(toJSON(root)).toMatchSnapshot();
    });

    it('should render 64 squares in total', () => {
        const root = mount(<Board />);
        const boardSquares = root.find(BoardSquare);
        expect(boardSquares.length).toEqual(64);
    });

    it('should swap two squares correctly', () => {
        const root = mount(<Board />);
        const boardSquares = root.find(BoardSquare);
        const mockSourcePostition = {
            x: 0,
            y: 0,
        };
        const mockTargetPosition = {
            x: 1,
            y: 2,
        };
        const sourceColor = boardSquares.at(0).props().color;
        const targetColor = boardSquares.at(10).props().color;
        root.find(BoardSquare).at(10).props().swapSquare(mockSourcePostition, mockTargetPosition);
        root.update();
        expect(root.find(BoardSquare).at(0).props().color).toEqual(targetColor);
        expect(root.find(BoardSquare).at(10).props().color).toEqual(sourceColor);
    });
})