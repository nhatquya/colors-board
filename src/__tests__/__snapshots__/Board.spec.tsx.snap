// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render board squares correctly 1`] = `
<Board>
  <div
    style={
      Object {
        "display": "flex",
        "flexWrap": "wrap",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Component
      color="#FFB3B3"
      key="x-0-y0"
      position={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FFB3B3"
        position={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FFDAB3"
      key="x-0-y1"
      position={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FFDAB3"
        position={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FFFFB3"
      key="x-0-y2"
      position={
        Object {
          "x": 0,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FFFFB3"
        position={
          Object {
            "x": 0,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#DAFFB3"
      key="x-0-y3"
      position={
        Object {
          "x": 0,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#DAFFB3"
        position={
          Object {
            "x": 0,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#B3FFB3"
      key="x-0-y4"
      position={
        Object {
          "x": 0,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#B3FFB3"
        position={
          Object {
            "x": 0,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#B3FFDA"
      key="x-0-y5"
      position={
        Object {
          "x": 0,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#B3FFDA"
        position={
          Object {
            "x": 0,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#B3FFFF"
      key="x-0-y6"
      position={
        Object {
          "x": 0,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#B3FFFF"
        position={
          Object {
            "x": 0,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#B3DAFF"
      key="x-0-y7"
      position={
        Object {
          "x": 0,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#B3DAFF"
        position={
          Object {
            "x": 0,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FF6B6B"
      key="x-1-y0"
      position={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FF6B6B"
        position={
          Object {
            "x": 1,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FFB66B"
      key="x-1-y1"
      position={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FFB66B"
        position={
          Object {
            "x": 1,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FFFF6B"
      key="x-1-y2"
      position={
        Object {
          "x": 1,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FFFF6B"
        position={
          Object {
            "x": 1,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#B6FF6B"
      key="x-1-y3"
      position={
        Object {
          "x": 1,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#B6FF6B"
        position={
          Object {
            "x": 1,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#6BFF6B"
      key="x-1-y4"
      position={
        Object {
          "x": 1,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#6BFF6B"
        position={
          Object {
            "x": 1,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#6BFFB6"
      key="x-1-y5"
      position={
        Object {
          "x": 1,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#6BFFB6"
        position={
          Object {
            "x": 1,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#6BFFFF"
      key="x-1-y6"
      position={
        Object {
          "x": 1,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#6BFFFF"
        position={
          Object {
            "x": 1,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#6BB6FF"
      key="x-1-y7"
      position={
        Object {
          "x": 1,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#6BB6FF"
        position={
          Object {
            "x": 1,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FF2F2F"
      key="x-2-y0"
      position={
        Object {
          "x": 2,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FF2F2F"
        position={
          Object {
            "x": 2,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FF982F"
      key="x-2-y1"
      position={
        Object {
          "x": 2,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FF982F"
        position={
          Object {
            "x": 2,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FFFF2F"
      key="x-2-y2"
      position={
        Object {
          "x": 2,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FFFF2F"
        position={
          Object {
            "x": 2,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#98FF2F"
      key="x-2-y3"
      position={
        Object {
          "x": 2,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#98FF2F"
        position={
          Object {
            "x": 2,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#2FFF2F"
      key="x-2-y4"
      position={
        Object {
          "x": 2,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#2FFF2F"
        position={
          Object {
            "x": 2,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#2FFF98"
      key="x-2-y5"
      position={
        Object {
          "x": 2,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#2FFF98"
        position={
          Object {
            "x": 2,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#2FFFFF"
      key="x-2-y6"
      position={
        Object {
          "x": 2,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#2FFFFF"
        position={
          Object {
            "x": 2,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#2F98FF"
      key="x-2-y7"
      position={
        Object {
          "x": 2,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#2F98FF"
        position={
          Object {
            "x": 2,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FC0000"
      key="x-3-y0"
      position={
        Object {
          "x": 3,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FC0000"
        position={
          Object {
            "x": 3,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FC7F00"
      key="x-3-y1"
      position={
        Object {
          "x": 3,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FC7F00"
        position={
          Object {
            "x": 3,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#FCFC00"
      key="x-3-y2"
      position={
        Object {
          "x": 3,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#FCFC00"
        position={
          Object {
            "x": 3,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#7FFC00"
      key="x-3-y3"
      position={
        Object {
          "x": 3,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#7FFC00"
        position={
          Object {
            "x": 3,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00FC00"
      key="x-3-y4"
      position={
        Object {
          "x": 3,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00FC00"
        position={
          Object {
            "x": 3,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00FC7F"
      key="x-3-y5"
      position={
        Object {
          "x": 3,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00FC7F"
        position={
          Object {
            "x": 3,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00FCFC"
      key="x-3-y6"
      position={
        Object {
          "x": 3,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00FCFC"
        position={
          Object {
            "x": 3,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#007FFC"
      key="x-3-y7"
      position={
        Object {
          "x": 3,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#007FFC"
        position={
          Object {
            "x": 3,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#D20000"
      key="x-4-y0"
      position={
        Object {
          "x": 4,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#D20000"
        position={
          Object {
            "x": 4,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#D26A00"
      key="x-4-y1"
      position={
        Object {
          "x": 4,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#D26A00"
        position={
          Object {
            "x": 4,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#D2D200"
      key="x-4-y2"
      position={
        Object {
          "x": 4,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#D2D200"
        position={
          Object {
            "x": 4,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#6AD200"
      key="x-4-y3"
      position={
        Object {
          "x": 4,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#6AD200"
        position={
          Object {
            "x": 4,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00D200"
      key="x-4-y4"
      position={
        Object {
          "x": 4,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00D200"
        position={
          Object {
            "x": 4,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00D26A"
      key="x-4-y5"
      position={
        Object {
          "x": 4,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00D26A"
        position={
          Object {
            "x": 4,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00D2D2"
      key="x-4-y6"
      position={
        Object {
          "x": 4,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00D2D2"
        position={
          Object {
            "x": 4,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#006AD2"
      key="x-4-y7"
      position={
        Object {
          "x": 4,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#006AD2"
        position={
          Object {
            "x": 4,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#AF0000"
      key="x-5-y0"
      position={
        Object {
          "x": 5,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#AF0000"
        position={
          Object {
            "x": 5,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#AF5800"
      key="x-5-y1"
      position={
        Object {
          "x": 5,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#AF5800"
        position={
          Object {
            "x": 5,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#AFAF00"
      key="x-5-y2"
      position={
        Object {
          "x": 5,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#AFAF00"
        position={
          Object {
            "x": 5,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#58AF00"
      key="x-5-y3"
      position={
        Object {
          "x": 5,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#58AF00"
        position={
          Object {
            "x": 5,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00AF00"
      key="x-5-y4"
      position={
        Object {
          "x": 5,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00AF00"
        position={
          Object {
            "x": 5,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00AF58"
      key="x-5-y5"
      position={
        Object {
          "x": 5,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00AF58"
        position={
          Object {
            "x": 5,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#00AFAF"
      key="x-5-y6"
      position={
        Object {
          "x": 5,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#00AFAF"
        position={
          Object {
            "x": 5,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#0058AF"
      key="x-5-y7"
      position={
        Object {
          "x": 5,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#0058AF"
        position={
          Object {
            "x": 5,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#920000"
      key="x-6-y0"
      position={
        Object {
          "x": 6,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#920000"
        position={
          Object {
            "x": 6,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#924900"
      key="x-6-y1"
      position={
        Object {
          "x": 6,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#924900"
        position={
          Object {
            "x": 6,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#929200"
      key="x-6-y2"
      position={
        Object {
          "x": 6,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#929200"
        position={
          Object {
            "x": 6,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#499200"
      key="x-6-y3"
      position={
        Object {
          "x": 6,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#499200"
        position={
          Object {
            "x": 6,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#009200"
      key="x-6-y4"
      position={
        Object {
          "x": 6,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#009200"
        position={
          Object {
            "x": 6,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#009249"
      key="x-6-y5"
      position={
        Object {
          "x": 6,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#009249"
        position={
          Object {
            "x": 6,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#009292"
      key="x-6-y6"
      position={
        Object {
          "x": 6,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#009292"
        position={
          Object {
            "x": 6,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#004992"
      key="x-6-y7"
      position={
        Object {
          "x": 6,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#004992"
        position={
          Object {
            "x": 6,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#7A0000"
      key="x-7-y0"
      position={
        Object {
          "x": 7,
          "y": 0,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#7A0000"
        position={
          Object {
            "x": 7,
            "y": 0,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#7A3D00"
      key="x-7-y1"
      position={
        Object {
          "x": 7,
          "y": 1,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#7A3D00"
        position={
          Object {
            "x": 7,
            "y": 1,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#7A7A00"
      key="x-7-y2"
      position={
        Object {
          "x": 7,
          "y": 2,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#7A7A00"
        position={
          Object {
            "x": 7,
            "y": 2,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#3D7A00"
      key="x-7-y3"
      position={
        Object {
          "x": 7,
          "y": 3,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#3D7A00"
        position={
          Object {
            "x": 7,
            "y": 3,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#007A00"
      key="x-7-y4"
      position={
        Object {
          "x": 7,
          "y": 4,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#007A00"
        position={
          Object {
            "x": 7,
            "y": 4,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#007A3D"
      key="x-7-y5"
      position={
        Object {
          "x": 7,
          "y": 5,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#007A3D"
        position={
          Object {
            "x": 7,
            "y": 5,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#007A7A"
      key="x-7-y6"
      position={
        Object {
          "x": 7,
          "y": 6,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#007A7A"
        position={
          Object {
            "x": 7,
            "y": 6,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
    <Component
      color="#003D7A"
      key="x-7-y7"
      position={
        Object {
          "x": 7,
          "y": 7,
        }
      }
      swapSquare={[Function]}
    >
      <mock-board-square
        color="#003D7A"
        position={
          Object {
            "x": 7,
            "y": 7,
          }
        }
        swapSquare={[Function]}
      />
    </Component>
  </div>
</Board>
`;
